conversation/promos-client
# [TC-959] Refactor placement engine to use promises

This PR refactors the placement engine to use plain old promises, rather than signals.

I originally chose signals for the architecture, because I thought that the placement engine needed to be reactive. It turns out that most of the reactive functionality ended up being removed, so signals just became cognitive overhead. Now that we are using promises, everything is much simpler.

I also shuffled a few things between the `placementEngine` and `placePromos` functions to make the API a bit clearer.

## How to Test

I have already [created a PR](https://github.com/conversation/tc/pull/8952) to upgrade to the new promos-client in TC. Fire it up and check that the promos all still work as intended.
